src/DynamoManipulation/NodeManipulator.cs

```diff
@@ -547,7 +548,11 @@ public void Dispose()
         {
             Dispose(true);
 
-            Node.ClearErrorsAndWarnings();
+            if (!string.IsNullOrEmpty(persistentWarning))
```

Comment:
do you think this deserves a comment why the disposal of a manipulator is clearing warnings?